File: 2028 Find Missing Observations/2028.cpp
class Solution {
public:
    vector<int> missingRolls(vector<int>& rolls, int mean, int n) {
        int sum = (n + rolls.size()) * mean - accumulate(rolls.begin(), rolls.end(), 0);
        if (sum < n || sum> 6 * n) return {};
        int q = sum / n, r = sum % n; 
        vector<int> ans(n, q);
        for (int i = 0; i < r; i++) ans[i]++;
        return ans;
    }
};

File: 2028 Find Missing Observations/2028.py
class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        total = (n + len(rolls)) * mean - sum(rolls)
        if total < n or total > 6 * n:
            return []
        else:
            q, r = total // n, total % n
            return [q + 1 if i < r else q for i in range(n)]
